src/nemos/base_regressor.py

```diff
@@ -45,6 +56,75 @@ def wrapper(self, *args, **kwargs):
     return decorator
 
 
+def get_name(x: object) -> str:
+    """
+    Get the name of an object ``x``, for saving/loading purposes.
+
+    Parameters
+    ----------
+    x
+        A python object or function.
+
+    Returns
+    -------
+    name
+        The name of that object, with full module path (e.g.,
+        ``nemos.observation_models.PoissonObservations``).
+    """
+    if x is None:
+        return None
+    try:
+        # if this passes, attr is a function
+        name = f"{x.__module__}.{x.__name__}"
+    except AttributeError:
+        # if we're here, then it's an object
+        cls = x.__class__
+        name = f"{cls.__module__}.{cls.__name__}"
+    return name
+
+
+def unpack_params(params_dict: dict) -> dict:
+    """
+    Convert a parameter dictionary into serializable format.
+
+    For objects with `get_params`/`set_params`, extracts the class name and
+    parameters. Some attributes are converted to strings to facilitate saving and loading.
+
+    Parameters
+    ----------
+    params_dict
+        Dictionary of parameters, possibly containing objects.
+
+    Returns
+    -------
+    dict
+        Serializable dictionary with class names and nested parameters.
+    """
+
+    attributes_as_string = ["inverse_link_function"]
```

Comment:
This should be an argument to the function -- it's something that may change with different subclasses.